File: source/net/auoeke/cin/lexer/lexeme/FloatLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class FloatLexeme extends Lexeme {
    public final String string;
    public final double value;

    public FloatLexeme(int line, int column, double value, String string, SyntaxError error) {
        super(line, column, error);

        this.value = value;
        this.string = string;
    }

    public FloatLexeme(int line, int column, double value, String string) {
        this(line, column, value, string, null);
    }

    @Override public Type type() {
        return Type.FLOAT;
    }

    @Override public String toString() {
        return this.string;
    }
}

File: src/net/auoeke/cin/lexer/lexeme/FloatLexeme.kt
package net.auoeke.cin.lexer.lexeme

import net.auoeke.extensions.*

class FloatLexeme(line: Int, column: Int, val value: Double, val integral: Boolean) : Lexeme(line, column) {
    override val type: Type get() = Type.FLOAT

    override fun toString(): String = value.string.mapIf(integral) {it.substring(0, it.indexOf('.') + 1)}
}
